src/panels/lovelace/editor/hui-dialog-edit-card.ts

```diff
@@ -59,20 +67,31 @@ export class HuiDialogEditCard extends LitElement {
         paper-dialog {
           width: 650px;
         }
+        .element-editor {
+          margin-bottom: 16px;
+        }
       </style>
       <paper-dialog with-backdrop>
         <h2>Card Configuration</h2>
         <paper-dialog-scrollable>
-          <hui-yaml-editor
-            .yaml="${this._cardConfig}"
-            @yaml-changed="${this._handleYamlChanged}"
-          ></hui-yaml-editor>
+          ${
+            this._showUIEditor
+              ? html`<div class="element-editor">${this._elementConfig}</div>`
```

Comment:
editor elements will need to fire a `@config-changed` event. Behaves similar to `yaml-changed`

Reply:
We should store the config as an instance variable as an object:

```ts
type ConfigValue = {
  format: string;
  value: string;
}
```

and we pass that to the preview function.

Reply:
May have to explain this one a bit more...

Reply:
Well we can either have a YAML based config or a config stored in JS objects (which we will render to JSON before saving). We need to know what format it is in when we send it to the backend and when we send it to the preview card. And if we don't want to have 2 different code paths, we need to keep track of the format the config is in.